synapse/api/auth.py

```diff
@@ -646,6 +677,14 @@ def compute_auth_events(self, event, current_state):
             if e_type == Membership.JOIN:
                 if member_event and not is_public:
                     auth_ids.append(member_event.event_id)
+                if ThirdPartyInvites.join_has_third_party_invite(event.content):
+                    key = (
+                        EventTypes.ThirdPartyInvite,
+                        event.content["third_party_invite"]["token"]
+                    )
+                    invite = current_state.get(key)
+                    if invite:
+                        auth_ids.append(invite.event_id)
```

Comment:
Nope; this is just a logic method to tell you whether there is a thing being satisfied; the caller is responsible for handling that how it will.